packages/@aws-cdk/aws-rds/lib/cluster.ts

```diff
@@ -340,6 +349,7 @@ abstract class DatabaseClusterNew extends DatabaseClusterBase {
     const clusterParameterGroup = props.parameterGroup ?? clusterEngineBindConfig.parameterGroup;
     const clusterParameterGroupConfig = clusterParameterGroup?.bindToCluster({});
     this.engine = props.engine;
+    this.enableIamAuthentication = props.iamAuthentication;
```

Comment:
Same question here - why is this new code in `DatabaseClusterNew`? I'd rather have it in `DatabaseCluster`, unless we're 100% certain this works for Clusters created from Snapshots too.